[
  {
    "project": "hyperledger/aries-cloudagent-python",
    "commit": "adbaa42736f88899c7aae15e20b4d9a261ae5178",
    "filename": "aries_cloudagent/config/injection_context.py",
    "filename_after_commit": "/home/chowyi/TypeAnnotation_Study/GitHub/hyperledger-aries-cloudagent-python/aries_cloudagent/config/injection_context.py",
    "file_hunks_size": 1,
    "min_patch_found": false,
    "single_hunk": false,
    "fit_TFix": false,
    "delete_only_patch": false,
    "has_suppression_all_hunks": false,
    "full_warning_msg": "aries_cloudagent/config/injection_context.py:108:4 Inconsistent override [14]: `aries_cloudagent.config.injection_context.InjectionContext.inject` overrides method defined in `BaseInjector` inconsistently. Could not find parameter `required` in overriding signature.",
    "message": " `aries_cloudagent.config.injection_context.InjectionContext.inject` overrides method defined in `BaseInjector` inconsistently. Could not find parameter `required` in overriding signature.",
    "rule_id": "Inconsistent override [14]",
    "warning_line_no": 108,
    "warning_line": "    def inject("
  },
  {
    "project": "hyperledger/aries-cloudagent-python",
    "commit": "adbaa42736f88899c7aae15e20b4d9a261ae5178",
    "filename": "aries_cloudagent/config/injector.py",
    "min_patch_found": false,
    "full_warning_msg": "aries_cloudagent/config/injector.py:93:4 Inconsistent override [14]: `aries_cloudagent.config.injector.Injector.inject` overrides method defined in `BaseInjector` inconsistently. Could not find parameter `required` in overriding signature.",
    "exception": "PatchSet should only have 1 Patch, but it has 0",
    "dd_fail": true
  },
  {
    "project": "hyperledger/aries-cloudagent-python",
    "commit": "adbaa42736f88899c7aae15e20b4d9a261ae5178",
    "filename": "aries_cloudagent/connections/base_manager.py",
    "min_patch_found": false,
    "full_warning_msg": "aries_cloudagent/connections/base_manager.py:169:8 Incompatible variable type [9]: storage is declared to have type `BaseStorage` but is used as type `typing.Optional[BaseStorage]`.",
    "exception": "PatchSet should only have 1 Patch, but it has 0",
    "dd_fail": true
  },
  {
    "project": "hyperledger/aries-cloudagent-python",
    "commit": "adbaa42736f88899c7aae15e20b4d9a261ae5178",
    "filename": "aries_cloudagent/connections/models/conn_record.py",
    "min_patch_found": false,
    "full_warning_msg": "aries_cloudagent/connections/models/conn_record.py:396:8 Incompatible variable type [9]: storage is declared to have type `BaseStorage` but is used as type `typing.Optional[BaseStorage]`.",
    "exception": "PatchSet should only have 1 Patch, but it has 0",
    "dd_fail": true
  },
  {
    "project": "hyperledger/aries-cloudagent-python",
    "commit": "adbaa42736f88899c7aae15e20b4d9a261ae5178",
    "filename": "aries_cloudagent/connections/models/conn_record.py",
    "min_patch_found": false,
    "full_warning_msg": "aries_cloudagent/connections/models/conn_record.py:409:8 Incompatible variable type [9]: storage is declared to have type `BaseStorage` but is used as type `typing.Optional[BaseStorage]`.",
    "exception": "PatchSet should only have 1 Patch, but it has 0",
    "dd_fail": true
  },
  {
    "project": "hyperledger/aries-cloudagent-python",
    "commit": "adbaa42736f88899c7aae15e20b4d9a261ae5178",
    "filename": "aries_cloudagent/connections/models/conn_record.py",
    "min_patch_found": false,
    "full_warning_msg": "aries_cloudagent/connections/models/conn_record.py:478:8 Incompatible variable type [9]: storage is declared to have type `BaseStorage` but is used as type `typing.Optional[BaseStorage]`.",
    "exception": "PatchSet should only have 1 Patch, but it has 0",
    "dd_fail": true
  },
  {
    "project": "hyperledger/aries-cloudagent-python",
    "commit": "adbaa42736f88899c7aae15e20b4d9a261ae5178",
    "filename": "aries_cloudagent/connections/models/conn_record.py",
    "min_patch_found": false,
    "full_warning_msg": "aries_cloudagent/connections/models/conn_record.py:498:8 Incompatible variable type [9]: storage is declared to have type `BaseStorage` but is used as type `typing.Optional[BaseStorage]`.",
    "exception": "PatchSet should only have 1 Patch, but it has 0",
    "dd_fail": true
  },
  {
    "project": "hyperledger/aries-cloudagent-python",
    "commit": "adbaa42736f88899c7aae15e20b4d9a261ae5178",
    "filename": "aries_cloudagent/connections/models/conn_record.py",
    "min_patch_found": false,
    "full_warning_msg": "aries_cloudagent/connections/models/conn_record.py:521:8 Incompatible variable type [9]: storage is declared to have type `BaseStorage` but is used as type `typing.Optional[BaseStorage]`.",
    "exception": "PatchSet should only have 1 Patch, but it has 0",
    "dd_fail": true
  },
  {
    "project": "hyperledger/aries-cloudagent-python",
    "commit": "adbaa42736f88899c7aae15e20b4d9a261ae5178",
    "filename": "aries_cloudagent/connections/models/conn_record.py",
    "min_patch_found": false,
    "full_warning_msg": "aries_cloudagent/connections/models/conn_record.py:542:8 Incompatible variable type [9]: storage is declared to have type `BaseStorage` but is used as type `typing.Optional[BaseStorage]`.",
    "exception": "PatchSet should only have 1 Patch, but it has 0",
    "dd_fail": true
  },
  {
    "project": "hyperledger/aries-cloudagent-python",
    "commit": "adbaa42736f88899c7aae15e20b4d9a261ae5178",
    "filename": "aries_cloudagent/core/dispatcher.py",
    "min_patch_found": false,
    "full_warning_msg": "aries_cloudagent/core/dispatcher.py:233:8 Incompatible variable type [9]: registry is declared to have type `ProtocolRegistry` but is used as type `typing.Optional[ProtocolRegistry]`.",
    "exception": "PatchSet should only have 1 Patch, but it has 0",
    "dd_fail": true
  },
  {
    "project": "hyperledger/aries-cloudagent-python",
    "commit": "adbaa42736f88899c7aae15e20b4d9a261ae5178",
    "filename": "aries_cloudagent/messaging/credential_definitions/routes.py",
    "min_patch_found": false,
    "full_warning_msg": "aries_cloudagent/messaging/credential_definitions/routes.py:203:20 Incompatible parameter type [6]: Expected `IndyIssuer` for 1st positional only parameter to call `BaseLedger.create_and_send_credential_definition` but got `typing.Optional[IndyIssuer]`.",
    "exception": "PatchSet should only have 1 Patch, but it has 0",
    "dd_fail": true
  },
  {
    "project": "hyperledger/aries-cloudagent-python",
    "commit": "adbaa42736f88899c7aae15e20b4d9a261ae5178",
    "filename": "aries_cloudagent/messaging/schemas/routes.py",
    "min_patch_found": false,
    "full_warning_msg": "aries_cloudagent/messaging/schemas/routes.py:203:20 Incompatible parameter type [6]: Expected `IndyIssuer` for 1st positional only parameter to call `BaseLedger.create_and_send_schema` but got `typing.Optional[IndyIssuer]`.",
    "exception": "PatchSet should only have 1 Patch, but it has 0",
    "dd_fail": true
  },
  {
    "project": "hyperledger/aries-cloudagent-python",
    "commit": "adbaa42736f88899c7aae15e20b4d9a261ae5178",
    "filename": "aries_cloudagent/multitenant/manager.py",
    "min_patch_found": false,
    "full_warning_msg": "aries_cloudagent/multitenant/manager.py:461:8 Incompatible variable type [9]: wire_format is declared to have type `BaseWireFormat` but is used as type `Optional[BaseWireFormat]`.",
    "exception": "PatchSet should only have 1 Patch, but it has 0",
    "dd_fail": true
  },
  {
    "project": "hyperledger/aries-cloudagent-python",
    "commit": "adbaa42736f88899c7aae15e20b4d9a261ae5178",
    "filename": "aries_cloudagent/protocols/connections/v1_0/manager.py",
    "min_patch_found": false,
    "full_warning_msg": "aries_cloudagent/protocols/connections/v1_0/manager.py:712:75 Incompatible parameter type [6]: Expected `BaseWallet` for 3rd positional only parameter to call `aries_cloudagent.messaging.agent_message.AgentMessage.sign_field` but got `typing.Optional[BaseWallet]`.",
    "exception": "PatchSet should only have 1 Patch, but it has 0",
    "dd_fail": true
  },
  {
    "project": "hyperledger/aries-cloudagent-python",
    "commit": "adbaa42736f88899c7aae15e20b4d9a261ae5178",
    "filename": "aries_cloudagent/protocols/didexchange/v1_0/manager.py",
    "min_patch_found": false,
    "full_warning_msg": "aries_cloudagent/protocols/didexchange/v1_0/manager.py:442:56 Incompatible parameter type [6]: Expected `BaseWallet` for 1st positional only parameter to call `aries_cloudagent.messaging.decorators.attach_decorator.AttachDecoratorData.verify` but got `typing.Optional[BaseWallet]`.",
    "exception": "PatchSet should only have 1 Patch, but it has 0",
    "dd_fail": true
  },
  {
    "project": "hyperledger/aries-cloudagent-python",
    "commit": "adbaa42736f88899c7aae15e20b4d9a261ae5178",
    "filename": "aries_cloudagent/protocols/didexchange/v1_0/manager.py",
    "min_patch_found": false,
    "full_warning_msg": "aries_cloudagent/protocols/didexchange/v1_0/manager.py:714:48 Incompatible parameter type [6]: Expected `BaseWallet` for 1st positional only parameter to call `DIDXManager.verify_diddoc` but got `typing.Optional[BaseWallet]`.",
    "exception": "PatchSet should only have 1 Patch, but it has 0",
    "dd_fail": true
  },
  {
    "project": "hyperledger/aries-cloudagent-python",
    "commit": "adbaa42736f88899c7aae15e20b4d9a261ae5178",
    "filename": "aries_cloudagent/protocols/discovery/v1_0/handlers/query_handler.py",
    "min_patch_found": false,
    "full_warning_msg": "aries_cloudagent/protocols/discovery/v1_0/handlers/query_handler.py:22:8 Incompatible variable type [9]: registry is declared to have type `ProtocolRegistry` but is used as type `typing.Optional[ProtocolRegistry]`.",
    "exception": "PatchSet should only have 1 Patch, but it has 0",
    "dd_fail": true
  },
  {
    "project": "hyperledger/aries-cloudagent-python",
    "commit": "adbaa42736f88899c7aae15e20b4d9a261ae5178",
    "filename": "aries_cloudagent/protocols/discovery/v1_0/routes.py",
    "min_patch_found": false,
    "full_warning_msg": "aries_cloudagent/protocols/discovery/v1_0/routes.py:48:4 Incompatible variable type [9]: registry is declared to have type `ProtocolRegistry` but is used as type `typing.Optional[ProtocolRegistry]`.",
    "exception": "PatchSet should only have 1 Patch, but it has 0",
    "dd_fail": true
  },
  {
    "project": "hyperledger/aries-cloudagent-python",
    "commit": "adbaa42736f88899c7aae15e20b4d9a261ae5178",
    "filename": "aries_cloudagent/protocols/issue_credential/v2_0/formats/ld_proof/handler.py",
    "min_patch_found": false,
    "full_warning_msg": "aries_cloudagent/protocols/issue_credential/v2_0/formats/ld_proof/handler.py:574:12 Incompatible parameter type [6]: Expected `typing.Callable[[str, typing.Dict[typing.Any, typing.Any]], typing.Dict[typing.Any, typing.Any]]` for 3rd parameter `document_loader` to call `aries_cloudagent.vc.vc_ld.verify.verify_credential` but got `typing.Optional[aries_cloudagent.vc.ld_proofs.document_loader.DocumentLoader]`.",
    "exception": "PatchSet should only have 1 Patch, but it has 0",
    "dd_fail": true
  },
  {
    "project": "hyperledger/aries-cloudagent-python",
    "commit": "adbaa42736f88899c7aae15e20b4d9a261ae5178",
    "filename": "aries_cloudagent/protocols/out_of_band/v1_0/manager.py",
    "min_patch_found": false,
    "full_warning_msg": "aries_cloudagent/protocols/out_of_band/v1_0/manager.py:689:20 Incompatible parameter type [6]: Expected `IndyHolder` for 3rd parameter `holder` to call `indy_proof_req_preview2indy_requested_creds` but got `Optional[IndyHolder]`.",
    "exception": "PatchSet should only have 1 Patch, but it has 0",
    "dd_fail": true
  },
  {
    "project": "hyperledger/aries-cloudagent-python",
    "commit": "adbaa42736f88899c7aae15e20b4d9a261ae5178",
    "filename": "aries_cloudagent/protocols/present_proof/dif/pres_exch_handler.py",
    "min_patch_found": false,
    "full_warning_msg": "aries_cloudagent/protocols/present_proof/dif/pres_exch_handler.py:413:24 Incompatible parameter type [6]: Expected `typing.Callable[[str, typing.Dict[typing.Any, typing.Any]], typing.Dict[typing.Any, typing.Any]]` for 4th parameter `document_loader` to call `derive_credential` but got `Optional[aries_cloudagent.vc.ld_proofs.document_loader.DocumentLoader]`.",
    "exception": "PatchSet should only have 1 Patch, but it has 0",
    "dd_fail": true
  },
  {
    "project": "hyperledger/aries-cloudagent-python",
    "commit": "adbaa42736f88899c7aae15e20b4d9a261ae5178",
    "filename": "aries_cloudagent/protocols/present_proof/dif/pres_exch_handler.py",
    "min_patch_found": false,
    "full_warning_msg": "aries_cloudagent/protocols/present_proof/dif/pres_exch_handler.py:1219:20 Incompatible parameter type [6]: Expected `typing.Callable[[str, typing.Dict[typing.Any, typing.Any]], typing.Dict[typing.Any, typing.Any]]` for 4th parameter `document_loader` to call `sign_presentation` but got `Optional[aries_cloudagent.vc.ld_proofs.document_loader.DocumentLoader]`.",
    "exception": "PatchSet should only have 1 Patch, but it has 0",
    "dd_fail": true
  },
  {
    "project": "hyperledger/aries-cloudagent-python",
    "commit": "adbaa42736f88899c7aae15e20b4d9a261ae5178",
    "filename": "aries_cloudagent/protocols/present_proof/dif/pres_exch_handler.py",
    "min_patch_found": false,
    "full_warning_msg": "aries_cloudagent/protocols/present_proof/dif/pres_exch_handler.py:1238:24 Incompatible parameter type [6]: Expected `typing.Callable[[str, typing.Dict[typing.Any, typing.Any]], typing.Dict[typing.Any, typing.Any]]` for 4th parameter `document_loader` to call `sign_presentation` but got `Optional[aries_cloudagent.vc.ld_proofs.document_loader.DocumentLoader]`.",
    "exception": "PatchSet should only have 1 Patch, but it has 0",
    "dd_fail": true
  },
  {
    "project": "hyperledger/aries-cloudagent-python",
    "commit": "adbaa42736f88899c7aae15e20b4d9a261ae5178",
    "filename": "aries_cloudagent/protocols/present_proof/v1_0/handlers/presentation_request_handler.py",
    "min_patch_found": false,
    "full_warning_msg": "aries_cloudagent/protocols/present_proof/v1_0/handlers/presentation_request_handler.py:99:20 Incompatible parameter type [6]: Expected `IndyHolder` for 3rd parameter `holder` to call `indy_proof_req_preview2indy_requested_creds` but got `typing.Optional[IndyHolder]`.",
    "exception": "PatchSet should only have 1 Patch, but it has 0",
    "dd_fail": true
  },
  {
    "project": "hyperledger/aries-cloudagent-python",
    "commit": "adbaa42736f88899c7aae15e20b4d9a261ae5178",
    "filename": "aries_cloudagent/protocols/present_proof/v1_0/manager.py",
    "min_patch_found": false,
    "full_warning_msg": "aries_cloudagent/protocols/present_proof/v1_0/manager.py:140:12 Incompatible parameter type [6]: Expected `BaseLedger` for 4th parameter `ledger` to call `aries_cloudagent.indy.models.pres_preview.IndyPresPreview.indy_proof_request` but got `typing.Optional[BaseLedger]`.",
    "exception": "PatchSet should only have 1 Patch, but it has 0",
    "dd_fail": true
  },
  {
    "project": "hyperledger/aries-cloudagent-python",
    "commit": "adbaa42736f88899c7aae15e20b4d9a261ae5178",
    "filename": "aries_cloudagent/protocols/present_proof/v2_0/formats/dif/handler.py",
    "min_patch_found": false,
    "full_warning_msg": "aries_cloudagent/protocols/present_proof/v2_0/formats/dif/handler.py:374:16 Incompatible parameter type [6]: Expected `typing.Callable[[str, typing.Dict[typing.Any, typing.Any]], typing.Dict[typing.Any, typing.Any]]` for 3rd parameter `document_loader` to call `verify_presentation` but got `typing.Optional[aries_cloudagent.vc.ld_proofs.document_loader.DocumentLoader]`.",
    "exception": "PatchSet should only have 1 Patch, but it has 0",
    "dd_fail": true
  },
  {
    "project": "hyperledger/aries-cloudagent-python",
    "commit": "adbaa42736f88899c7aae15e20b4d9a261ae5178",
    "filename": "aries_cloudagent/protocols/present_proof/v2_0/formats/indy/handler.py",
    "min_patch_found": false,
    "full_warning_msg": "aries_cloudagent/protocols/present_proof/v2_0/formats/indy/handler.py:141:24 Incompatible parameter type [6]: Expected `IndyHolder` for 3rd parameter `holder` to call `indy_proof_req_preview2indy_requested_creds` but got `typing.Optional[IndyHolder]`.",
    "exception": "PatchSet should only have 1 Patch, but it has 0",
    "dd_fail": true
  },
  {
    "project": "hyperledger/aries-cloudagent-python",
    "commit": "adbaa42736f88899c7aae15e20b4d9a261ae5178",
    "filename": "aries_cloudagent/transport/inbound/manager.py",
    "min_patch_found": false,
    "full_warning_msg": "aries_cloudagent/transport/inbound/manager.py:168:12 Incompatible variable type [9]: wire_format is declared to have type `BaseWireFormat` but is used as type `typing.Optional[BaseWireFormat]`.",
    "exception": "PatchSet should only have 1 Patch, but it has 0",
    "dd_fail": true
  }
]